akka-stream/src/main/scala/akka/stream/impl/Throttle.scala

```diff
@@ -32,10 +32,17 @@ import scala.concurrent.duration.{ FiniteDuration, _ }
   // small which is usually at rates where that precision is highly unlikely anyway as the overhead of this stage
   // is likely higher than the required accuracy interval.
   private val nanosBetweenTokens = per.toNanos / cost
+
+  println(s"# millis between tokens: ${nanosBetweenTokens / 1000 / 1000}") // FIXME
+
   private val timerName: String = "ThrottleTimer"
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new TimerGraphStageLogic(shape) {
-    private val tokenBucket = new NanoTimeTokenBucket(maximumBurst, nanosBetweenTokens)
+    // 100 ms is a realistic minimum between tokens, otherwise the maximumBurst is adjusted
+    // to be able to support higher rates
+    val adjustedMaximumBurst = math.max(maximumBurst, ((100 * 1000 * 1000) / nanosBetweenTokens))
+    println(s"# adjustedMaximumBurst: $adjustedMaximumBurst, maximumBurst: $maximumBurst") // FIXME
```

Comment:
My main WIP question is if we should always do this adjustment or if we should only do it for `throttleEven`?

It might be most theoretically correct to not do it for `throttle` but it's very easy to get the parameters to `throttle` wrong and then the actual throttling rate limited by the scheduler overhead kicks in (and that is for very low rates).

Reply:
Can we provide a `throttle` overload without the parameter and adjust for that one?

Reply:
yes, and then deprecated `throttleEven` which is no different?

Reply:
Sounds good. I always found this confusing, having a single goto `throttle` definitely sounds like an user experience improvement. :)